qiskit_nature/transformers/active_space_transformer.py

```diff
@@ -0,0 +1,352 @@
+        beta = mo_coeff_full[1] is not None
+        mo_occ_total = mo_occ_full[0] + mo_occ_full[1] if beta else mo_occ_full[0]
+
+        # compute number of inactive electrons
+        nelec_total = q_molecule.num_alpha + q_molecule.num_beta
+        nelec_inactive = nelec_total - self.num_electrons
+        if self.num_alpha is not None:
+            if not beta:
+                warning = 'The provided instance of QMolecule does not provide any beta ' \
+                          + 'coefficients but you tried to specify a separate number of alpha ' \
+                          + 'electrons. Continuing as if it does not matter.'
+                logger.warning(warning)
+            num_alpha = self.num_alpha
+            num_beta = self.num_electrons - self.num_alpha
+        else:
+            num_beta = (self.num_electrons - (q_molecule.multiplicity - 1)) // 2
+            num_alpha = self.num_electrons - num_beta
+
+        self._validate_num_electrons(nelec_inactive)
+        self._validate_num_orbitals(nelec_inactive, q_molecule, mo_occ_total)
+
+        # determine active and inactive orbital indices
+        if self.active_orbitals is None:
+            norbs_inactive = nelec_inactive // 2
+            inactive_orbs_idxs = list(range(norbs_inactive))
+            active_orbs_idxs = list(range(norbs_inactive, norbs_inactive+self.num_orbitals))
+        else:
+            active_orbs_idxs = self.active_orbitals
+            inactive_orbs_idxs = [o for o in range(nelec_total // 2) if o not in
+                                  self.active_orbitals and mo_occ_total[o] > 0]
+
+        # split molecular orbitals coefficients into active and inactive parts
+        mo_coeff_inactive = (mo_coeff_full[0][:, inactive_orbs_idxs],
+                             mo_coeff_full[1][:, inactive_orbs_idxs] if beta else None)
+        mo_coeff_active = (mo_coeff_full[0][:, active_orbs_idxs],
+                           mo_coeff_full[1][:, active_orbs_idxs] if beta else None)
+        mo_occ_inactive = (mo_occ_full[0][inactive_orbs_idxs],
+                           mo_occ_full[1][inactive_orbs_idxs] if beta else None)
+
+        density_inactive = self._compute_inactive_density_matrix(mo_occ_inactive, mo_coeff_inactive)
+
+        # extract core Hamiltonian and electron-repulsion-integral matrices from QMolecule
+        hcore = (q_molecule.hcore, q_molecule.hcore_b if beta else None)
+        eri = q_molecule.eri
+
+        fock_inactive = self._compute_inactive_fock_op(hcore, eri, density_inactive)
+
+        e_inactive = self._compute_inactive_energy(hcore, density_inactive, fock_inactive,
+                                                   mo_coeff_inactive)
+
+        hij, hijkl = self._compute_active_integrals(mo_coeff_active, fock_inactive, eri)
+
+        # construct new QMolecule
+        q_molecule_reduced = copy.deepcopy(q_molecule)
+        # Energies and orbits
+        q_molecule_reduced.energy_shift['inactive_energy'] = e_inactive
```

Comment:
If we do this as string does it make sense to make it something more like `Active Space` in this case then when a result is generated if we want the extracted energies can be printed along the lines of `key + " energy extracted = " + value` for each entry and the naming allows user to correlate the extracted part (however we want to call it) to the transform. That way we can see the `Particle Hole` part etc etc and better know what they are.

Reply:
I think `Active Space` is a bit misleading since it would suggest the energy corresponds to the active part (although it's the opposite). I changed it to `ActiveSpaceTransformer` instead.